manifests/osfamily/redhat.pp

```diff
@@ -0,0 +1,45 @@
+class agent_upgrade::osfamily::redhat {
+  #  if $caller_module_name != $module_name {
+  #  fail("Use of private class ${name} by ${caller_module_name}")
+  #}
```

Comment:
Oh right, I was going to try the guard I saw in another repo. I'll check whether I want to enable it or remove it completely.